internal/provider/kubernetes/gateway.go

```diff
@@ -179,6 +286,8 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, request reconcile.Req
 	}
 
 	found := false
+	var secrets []corev1.Secret
+	var requestedGateway gwapiv1b1.Gateway
```

Comment:
I dont quite understand how `requestedGateway` is being used below in the `found` & `!found` logic